docs2/site/docs/migrations/migration5.md

```diff
@@ -415,6 +417,64 @@ servcies.AddGraphQL(builder => builder
 The `DocumentExecuter.SelectExecutionStrategy` method is still available to be overridden for
 backwards compatibility but may be removed in the next major version.
 
+### 13. Schema builder improvements for reflected methods
+
+When configuring a CLR method for a field, the method arguments now allow the use of all of
+the new attributes available to `AutoRegisteringObjectGraphType`, such as `[FromServices]`.
+Field resolvers are now precompiled, resulting in faster performance.
+
+As always, when the CLR type is not the source type, the CLR type is pulled from DI.
+Now the CLR type will be pulled from `context.RequestServices` to allow for scoped instances.
+If `RequestServices` is `null`, the root DI provider will be used as it was before.
+
+Note that existing methods will require the use of `[FromSource]` and `[FromUserContext]` for
+applicable method arguments.
+
+```csharp
+// v4
+[GraphQLMetadata("Droid")]
+class DroidType
+{
+    // DI-injected services are always pulled from the root DI provider, so scoped services are not supported
+    private readonly Repository _repo;
+    public DroidType(Repository repo)
+    {
+        _repo = repo;
+    }
+
+    public int Id(Droid source) => source.Id;
+
+    public IEnumerable<Droid> Friends(Droid source) => _repo.FriendsOf(source.Id);
+}
+
+// v5
+[GraphQLMetadata("Droid")]
+class DroidType
+{
+    // scoped services are supported, so long as ExecutionOptions.RequestServices is set
+    private readonly Repository _repo;
+    public DroidType(Repository repo)
+    {
+        _repo = repo;
+    }
+
+    // requires use of [FromSource]
+    public int Id([FromSource] Droid source) => source.Id;
+
+    public IEnumerable<Droid> Friends([FromSource] Droid source) => _repo.FriendsOf(source.Id);
+}
+
+// v5 alternate
+[GraphQLMetadata("Droid")]
+class DroidType
+{
+    public int Id([FromSource] Droid source) => source.Id;
```

Comment:
Of course, with the attribute, it is clear how to get value for method.

Reply:
`IsInstanceOfType` from existing code looks not very good. As well as whole `ReflectionHelper.BuildArguments`.

Reply:
`ReflectionHelper.BuildArguments` will be removed once I finish PRs to alter/remove the methods that depend on it.